File: frontend/src/interfaces/stage.ts
import type { StageEquipmentInterface } from "./stageequipmnt";
import type { StageTypeInterface } from "./stagetype";
import type { VenueInterface } from "./venue";

export interface StageInterface {
  id: number;
  stage_name?: string;
  width: number;
  length: number;
  stage_type_id: number;
  stage_type?: StageTypeInterface;
  venue_id: number;
  venue?: VenueInterface; // optional circular reference
  equipments?: StageEquipmentInterface[];

}

File: backend/entity/stage.go
package entity
import "gorm.io/gorm"

type Stage struct {
	gorm.Model
	StageName string `json:"stage_name"`
	Width float32 `json:"width"`
	Length float32 `json:"length"`

	StageTypeID uint `json:"stage_type_id"`
	StageType *StageType `gorm:"foreignKey: stage_type_id" json:"stage_type"`

	VenueID uint `json:"venue_id"`
	Venue *Venue `gorm:"foreignKey: venue_id" json:"venue"`
	
	Equipments []StageEquipment `gorm:"foreignKey:StageID" json:"equipments"`
} 